scipy/stats/_continuous_distns.py

```diff
@@ -3348,6 +3348,283 @@ def _stats(self, mu):
 invgauss = invgauss_gen(a=0.0, name='invgauss')
 
 
+class geninvgauss_gen(rv_continuous):
+    r"""A Generalized Inverse Gaussian continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `geninvgauss` is:
+
+    .. math::
+
+        f(x, p, b) = 1 / (2 K_p(b)) * (x^{p-1} \exp(-0.5 * b * (x + 1/x)))
+
+    where `x > 0`, and the parameters `p, b` satisfy `b`> 0` ([1]_).
+    :math:`K_p` is the modified Bessel function of second kind of order `p`
+    (`scipy.special.kv`).
+
+    %(after_notes)s
+
+    The inverse Gaussian distribution `stats.invgauss(mu)` is a special case of
+    `geninvgauss` with `p = -1/2`, `b = 1 / mu` and `scale = mu`.
+
+    Generating random variates is challenging for this distribution. The
+    implementation is based on [2]_.
+
+    References
+    ----------
+    .. [1] O. Barndorff-Nielsen, P. Blaesild, C. Halgreen, "First hitting time
+       models for the generalized inverse gaussian distribution",
+       Stochastic Processes and their Applications 7, pp. 49--54, 1978.
+
+    .. [2] W. Hoermann and J. Leydold, "Generating generalized inverse Gaussian
+       random variates", Statistics and Computing, 24(4), p. 547--557, 2014.
+
+    %(example)s
+
+    """
+    def _argcheck(self, p, b):
+        return (p == p) & (b > 0)
+
+    def _logpdf(self, x, p, b):
+        c = -np.log(2*sc.kv(p, b))
+        return _lazywhere(x > 0, (x, p, b, c),
+                          lambda x, p, b, c: c + (p-1)*np.log(x) - b*(x+1/x)/2,
```

Comment:
PEP 8: Change `p-1` to `p - 1`, and `x+1/x` to `x + 1/x`.  Then wrap the line to keep the line lengths 79 characters or less.